src/engine/build.bat

```diff
@@ -106,19 +106,19 @@ if NOT "_%VS150COMNTOOLS%_" == "__" (
     set "BOOST_JAM_TOOLSET_ROOT=%VS150COMNTOOLS%..\..\VC\"
     goto :eof)
 call :Clear_Error
-if EXIST "%ProgramFiles%\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat"  (
+if EXIST "%ProgramFiles(x86)%\Microsoft Visual Studio\2017\Enterprise\VC\Auxiliary\Build\vcvarsall.bat"  (
```

Comment:
I had thought that windows 32-bit (Vista and up) defined both %ProgramFiles% and %ProgramFiles(x86)% to be the same C:\Program Files\. Then on 64-bit systems, %ProgramFiles% points to C:\Program Files\ and %ProgramFiles(x86)% points to C:\Program Files (x86)\.  Is that not correct? I haven't had a 32-bit machine for a long time, so I can't easily look.

Regardless, I think searching for a specific path is a bad long-term solution. There isn't any guarantee that a user will install it there (it is selectable during install). If the VS15COMNTOOLS variable isn't reliable, then maybe we can query the registry or something to make it work? 

I really just submitted this pull request as an interm fix because it was breaking my regression tests on a machine that happened to have msvc-15 installed on it.